specs/cli/client-add.md

```diff
@@ -0,0 +1,125 @@
+
+## Parameters
+
+| Parameters | Required | Example | Description |
+| -- | -- | -- | -- |
+| `--config-location \| --cl` | No | ../../ | A location where to find or create the `kiota-config.json` file. When not specified it will find an ancestor `kiota-config.json` file and if not found, will use the defaults. Defaults to `./`. |
+| `--client-name \| --cn` | Yes | graphDelegated | Name of the client. Unique within the parent API. If not provided, defaults to --class-name or its default. |
+| `--openapi \| -d` | Yes | https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml | The location of the OpenAPI description in JSON or YAML format to use to generate the SDK. Accepts a URL or a local path. |
+| `--search-key \| --sk` | No | github::microsoftgraph/msgraph-metadata/graph.microsoft.com/v1.0 | The search key used to locate the OpenAPI description. |
+| `--include-path \| -i` | No | /me/chats#GET | A glob pattern to include paths from generation. Accepts multiple values. Defaults to no value which includes everything. |
+| `--exclude-path \| -e` | No | \*\*/users/\*\* | A glob pattern to exclude paths from generation. Accepts multiple values. Defaults to no value which excludes nothing. |
+| `--language \| -l` | Yes | csharp | The target language for the generated code files or for the information. |
+| `--class-name \| -c` | No | GraphClient | The name of the client class. Defaults to `Client`. |
+| `--namespace-name \| -n` | No | Contoso.GraphApp | The namespace of the client class. Defaults to `Microsoft.Graph`. |
+| `--backing-store \| -b` | No | | Defaults to `false` |
+| `--exclude-backward-compatible \| --ebc` | No |  | Whether to exclude the code generated only for backward compatibility reasons or not. Defaults to `false`. |
+| `--serializer \| -s` | No | `Contoso.Json.CustomSerializer` | One or more module names that implements ISerializationWriterFactory. Default are documented [here](https://learn.microsoft.com/openapi/kiota/using#--serializer--s). |
+| `--deserializer \| --ds` | No | `Contoso.Json.CustomDeserializer` | One or more module names that implements IParseNodeFactory. Default are documented [here](https://learn.microsoft.com/en-us/openapi/kiota/using#--deserializer---ds). |
+| `--structured-media-types \| -m` | No | `application/json` |Any valid media type which will match a request body type or a response type in the OpenAPI description. Default are documented [here](https://learn.microsoft.com/en-us/openapi/kiota/using#--structured-mime-types--m). |
+| `--skip-generation \| --sg` | No | true | When specified, the generation would be skipped. Defaults to false. |
+| `--output \| -o` | No | ./generated/graph/csharp | The output directory or file path for the generated code files. Defaults to `./output`. |
+
+> [!NOTE] 
```

Comment:
I just want to call out that if people do that, and with the current design, the only way they'll be able to get code to be generated/new edits to be taken into account for parameters, it with `generate --refresh`. Kiota is not going to be able to tell parameters changes, or whether the code has already been generated.

Reply:
`kiota client generate` should always regen, just not update the description. As the description will not have been cached, I feel this is a valid scenario and should fetch, generate, cache and store the hash in the apimanifest.json.

Reply:
just detecting the presence of artifacts will only partially solve the problem. What if the include/exclude patterns are edited manually? (without the command)

Reply:
That would be regenerated, no? `--refresh` should only update the cached description, nothing more, right?

Reply:
This is one of the things we've discussed in our last meeting. If we don't have a copy of all those settings in the manifest file, kiota can't tell they've been changed in the config file when running the generate command

Reply:
The API Manifest should not become bloated with all sorts of information. Here's a suggestion: Why aren't we generating a hash of the "resulting surface" instead of the entire OpenAPI file? That way, if what the consumer needs is not impacted by a newer of the OpenAPI description, we would not regen. In the case we're mentionmiong, we would detect that the resulting hash would be different, therefore would regen, re-cache the description, etc. I think this is the only scenario that sould land in the API Manifest and seems to solve these issues. Other manual edits should just get regen.

Reply:
Yep, a hash of all the parameters could work as well

Reply:
Ok, I think this can work. We'll have a hash of the entire json object for the client (including its name). I'll update the spec accordingly.

Reply:
The behavior was updated in the different commands.